index.d.ts

```diff
@@ -1,4 +1,10 @@
-export interface Token {
+declare var MarkdownIt: {
+  (preset?: string, options?: MarkdownIt.Options): MarkdownIt.MarkdownIt;
+  new (preset?: string, options?: MarkdownIt.Options): MarkdownIt.MarkdownIt;
+}
+
+declare namespace MarkdownIt {
```

Comment:
To also expose all the interfaces in use when doing `export =`, we use a namespace (yes, it's a little hacky, but it allows you to reuse the interfaces outside of the module properly).